File: app/src/main/java/com/pnuema/android/bible/ui/utils/DialogUtils.java
package com.pnuema.android.bible.ui.utils;

import androidx.fragment.app.FragmentActivity;

import com.pnuema.android.bible.ui.dialogs.BCVDialog;
import com.pnuema.android.bible.ui.dialogs.NotifySelectionCompleted;
import com.pnuema.android.bible.ui.dialogs.NotifyVersionSelectionCompleted;
import com.pnuema.android.bible.ui.dialogs.VersionSelectionDialog;

/**
 * Dialog utilities to generate and display dialogs.
 */
public final class DialogUtils {
    private DialogUtils() {
    }

    public static void showBookChapterVersePicker(final FragmentActivity activity, final BCVDialog.BCV bcv, final NotifySelectionCompleted listener) {
        BCVDialog.Companion.instantiate(bcv, listener).show(activity.getSupportFragmentManager(), BCVDialog.class.getSimpleName());
    }

    public static void showVersionsPicker(final FragmentActivity activity, final NotifyVersionSelectionCompleted listener) {
        VersionSelectionDialog.Companion.instantiate(listener).show(activity.getSupportFragmentManager(), VersionSelectionDialog.class.getSimpleName());
    }
}

File: app/src/main/java/com/pnuema/android/bible/ui/utils/DialogUtils.kt
package com.pnuema.android.bible.ui.utils

import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager

import com.pnuema.android.bible.ui.dialogs.BCVDialog
import com.pnuema.android.bible.ui.dialogs.NotifySelectionCompleted
import com.pnuema.android.bible.ui.dialogs.NotifyVersionSelectionCompleted
import com.pnuema.android.bible.ui.dialogs.VersionSelectionDialog

/**
 * Dialog utilities to generate and display dialogs.
 */
object DialogUtils {

    fun showBookChapterVersePicker(activity: FragmentActivity, bcv: BCVDialog.BCV, listener: NotifySelectionCompleted) {
        closeDialogs<VersionSelectionDialog>(activity.supportFragmentManager)
        BCVDialog.instantiate(bcv, listener).show(activity.supportFragmentManager, BCVDialog::class.java.simpleName)
    }

    fun showVersionsPicker(activity: FragmentActivity, listener: NotifyVersionSelectionCompleted) {
        closeDialogs<VersionSelectionDialog>(activity.supportFragmentManager)
        VersionSelectionDialog.instantiate(listener).show(activity.supportFragmentManager, VersionSelectionDialog::class.java.simpleName)
    }

    private inline fun <reified T : DialogFragment>closeDialogs(supportFragmentManager: FragmentManager) {
        supportFragmentManager.fragments.filterIsInstance<T>().forEach {
            it.dismiss()
        }
    }
}
